### PR TITLE
Small fixes.

### PR DESCRIPTION
**Pull request by nsoranzo**
- Mention SQLite dependency in README.md .
- Update sparsehash URL.
- Add many files to .gitignore .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,9 +69,10 @@ Dependencies
 
 ABySS requires the following libraries:
 
-* [Boost](http://www.boost.org)
-* [sparsehash](http://code.google.com/p/sparsehash)
+* [Boost](http://www.boost.org/)
 * [Open MPI](http://www.open-mpi.org)
+* [sparsehash](https://code.google.com/p/sparsehash/)
+* [SQLite](http://www.sqlite.org/)
```
Comment (sjackman) [inline README.md]: This dependency is optional. In fact, Boost is the only required dependency, but the others are heavily recommended. Should we also mention a compiler capable of OpenMP here?

**Comment by sjackman** [comment]
Thanks for your contribution, Nicola.

File: README.md (hunk 1 above)
Comment (nsoranzo) [inline README.md]: `make` fails for me without sqlite3:

```
Making all in DataBase
make[2]: Entering directory `/usr/users/ga002/soranzon/software/bcgsc_abyss/DataBase'
g++ -Wall -Wextra -g -O2   -o abyss-db-csv abyss_db_csv-DB.o abyss_db_csv-db-csv.o -lsqlite3 -ldl -lm 
/usr/bin/ld: cannot find -lsqlite3
collect2: error: ld returned 1 exit status
make[2]: *** [abyss-db-csv] Error 1
```

**Comment by benvvalk** [comment]
Ditto!

File: README.md (hunk 1 above)
Comment (nsoranzo) [inline README.md]: I think this is due to https://github.com/bcgsc/abyss/blob/master/DataBase/Makefile.am#L9

**Comment by sjackman** [comment]
Ben, sqlite3 should be an optional dependency. Can you fix this?

**Pull request merged**